backend/src/main/java/com/github/redstylzz/backend/service/CategoryService.java

```diff
@@ -72,14 +76,23 @@ private boolean categoryExistent(String name) {
             LOG.debug("Category does not exist");
             throw new CategoryDoesNotExistException("No category with ID: " + categoryID);
         }
-        return repository.findAllByUserID(user.getId());
+        return categoryRepository.findAllByUserID(user.getId());
     }
 
-    // TODO Remove associated payments
     public List<Category> deleteCategory(MongoUser user, String categoryID) {
         LOG.debug("Deleting category with ID: " + categoryID + " from user: " + user.getUsername());
-        repository.deleteByCategoryID(categoryID);
-        return repository.findAllByUserID(user.getId());
+        categoryRepository.deleteByCategoryID(categoryID);
+        paymentRepo.deleteAllByUserIDAndCategoryID(user.getId(), categoryID);
+        return categoryRepository.findAllByUserID(user.getId());
+    }
+
+    public void setCategorySum(String userID, String categoryID, BigDecimal paymentSum) {
+        LOG.debug("Calculating payment sum for category: " + categoryID);
```

Comment:
this log feels misleading - you are not calculating a payment sum, you are simply updating it, your log should ideally reflect that